components/groups/read/group-summary-view/group-summary-view.tsx

```diff
@@ -1,76 +1,149 @@
-import DynamicFieldView from '@components/generic/dynamic-field/dynamic-field-view'
+import React, { memo, useCallback, useEffect, useState } from 'react'
+import Link from 'next/link'
+
 import { FIELDS } from '@lib/constants'
+import fetch from '@lib/fetcher'
+import { useUser } from '@lib/client/hooks'
+import { groupUserPermission } from '@lib/permissions'
+import API from '@lib/api/api-endpoints'
+import { GroupEntityType, GroupTypeDefinition } from '@lib/types'
+import DynamicFieldView from '@components/generic/dynamic-field/dynamic-field-view'
 import GroupMembers from '@components/groups/group-members/group-members'
-import Link from 'next/link'
+import Button from '@components/generic/button/button'
 
-export default function(props) {
-  const shouldAddLink = (field) => {
-    return (
-      props.linkToDetail &&
-      field.type !== FIELDS.IMAGE &&
-      field.type !== FIELDS.FILE &&
-      field.type !== FIELDS.TAGS &&
-      field.type !== FIELDS.VIDEO_URL
-    )
-  }
+interface Props {
+  linkToDetail: boolean
+  group: GroupEntityType
+  className: string
+  type: GroupTypeDefinition
+}
+
+function SummaryView({ linkToDetail, group, className, type }: Props) {
+  const [clicked, setClicked] = useState<boolean>(false)
+  const [error, setError] = useState<boolean>(false)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [inMembersList, setInMembersList] = useState<boolean>(false)
+  const [inPendingMembersList, setInPendingMembersList] = useState<boolean>(
+    false
+  )
+  const shouldAddLink = useCallback(
+    (field) => {
+      return (
+        linkToDetail &&
+        ![FIELDS.IMAGE, FIELDS.FILE, FIELDS.TAGS, FIELDS.VIDEO_URL].includes(
+          field.type
+        )
+      )
+    },
+    [linkToDetail]
+  )
+
+  const currentUser = useUser()
+  const canJoin = groupUserPermission(
+    currentUser?.user,
+    type.slug,
+    'join',
+    group
+  )
+
+  useEffect(() => {
+    if (currentUser.user) {
+      const {
```

Comment:
do not update the client data, rather, refetch the data from the API .

Reply:
I am not updating anything there. these are just markers